### PR TITLE
[tests-only][full-ci]Cache playwright chromium

### PR DESCRIPTION
**Pull request by amrita-shrestha**
## Description
This PR 
- update `PLUGINS_S3 = "plugins/s3:1.5"` version
- cache chromium with respect to playwright version
- Use cached chromium to run e2e tests
- add script to get playwright version to store cache 

## Note
- can't upgrade minio version because some command are not available in the latest minio image 

## Related Issue

- Fixes https://github.com/owncloud/web/issues/11921

## How Has This Been Tested?

- CI

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Technical debt
- [ ] Tests
- [ ] Documentation
- [ ] Maintenance (e.g. dependency updates or tooling)

## Open tasks:

- [ ] ...

### CODE REVIEW

File: .drone.star [hunk 1]
```diff
@@ -1899,3 +1908,40 @@ def getOcislatestCommitId(ctx):
...
+    return [{
+        "name": "upload-chromium-cache",
+        "image": MINIO_MC,
+        "environment": minio_mc_environment,
+        "commands": [
+            "playwright_version=$(bash tests/drone/script.sh get_playwright_version)",
+            "mc alias set s3 $MC_HOST $AWS_ACCESS_KEY_ID $AWS_SECRET_ACCESS_KEY",
+            "mc cp -r -a %s/.playwright s3/$CACHE_BUCKET/web/$playwright_version/" % dir["web"],
+            "mc ls --recursive s3/$CACHE_BUCKET/web",
+        ],
```
Comment (saw-jan) [inline .drone.star:1922]: we should zip the `.playwright` folder and push the archived file to the cache
See https://github.com/owncloud/ocis/blob/46d397d74ff8cda2197746fed3708eab9f34d13f/.drone.star#L2966-L2985

File: .drone.star [hunk 2]
```diff
@@ -1899,3 +1908,40 @@ def getOcislatestCommitId(ctx):
...
+def restoreChromiumCache():
+    return [{
+        "name": "restore-chromium-cache",
+        "image": MINIO_MC,
+        "environment": minio_mc_environment,
+        "commands": [
+            "playwright_version=$(bash tests/drone/script.sh get_playwright_version)",
+            "mc alias set s3 $MC_HOST $AWS_ACCESS_KEY_ID $AWS_SECRET_ACCESS_KEY",
+            "mc cp -r -a s3/$CACHE_BUCKET/web/$playwright_version/.playwright %s" % dir["web"],
+        ],
```
Comment (saw-jan) [inline .drone.star:1946]: and restore the archived file and then unzip to the desired location
See https://github.com/owncloud/ocis/blob/46d397d74ff8cda2197746fed3708eab9f34d13f/.drone.star#L3016-L3032

File: .drone.star [hunk 3]
```diff
@@ -39,6 +39,7 @@ dir = {
     "ocisRevaDataRoot": "/srv/app/tmp/ocis/owncloud/data/",
     "federatedOcisConfig": "/var/www/owncloud/web/tests/drone/config-ocis-federated.json",
     "ocmProviders": "/var/www/owncloud/web/tests/drone/providers.json",
+    "chromiumZip": "/var/www/owncloud/web/playwright-chromium.tar.gz",
```
Comment (saw-jan) [inline .drone.star:42]:
```suggestion
    "playwrightBrowsersArchive": "/var/www/owncloud/web/playwright-browsers.tar.gz",
```

File: .drone.star [hunk 4]
```diff
@@ -716,15 +723,16 @@ def installPnpm():
...
 
-def installPlaywright():
+def installChromium():
     return [{
-        "name": "playwright-install",
+        "name": "chromium-install",
         "image": OC_CI_NODEJS,
         "environment": {
-            "PLAYWRIGHT_BROWSERS_PATH": ".playwright",
+            "PLAYWRIGHT_BROWSERS_PATH": ".chromium",
```
Comment (saw-jan) [inline .drone.star:731]:
```suggestion
            "PLAYWRIGHT_BROWSERS_PATH": ".playwright",
```

File: .drone.star [hunk 5]
```diff
@@ -716,15 +723,16 @@ def installPnpm():
         ],
     }]
 
-def installPlaywright():
+def installChromium():
```
Comment (saw-jan) [inline .drone.star:726]:
```suggestion
def installBrowsers():
```

File: .drone.star [hunk 6]
```diff
@@ -716,15 +723,16 @@ def installPnpm():
...
     return [{
-        "name": "playwright-install",
+        "name": "chromium-install",
```
Comment (saw-jan) [inline .drone.star:728]:
```suggestion
        "name": "install-browsers",
```

File: .drone.star [hunk 7]
```diff
@@ -1899,3 +1910,51 @@ def getOcislatestCommitId(ctx):
             ],
         },
     ]
+
+def cacheChromium():
```
Comment (saw-jan) [inline .drone.star:1914]:
```suggestion
def cacheBrowsers():
```

File: .drone.star [hunk 8]
```diff
@@ -1899,3 +1910,51 @@ def getOcislatestCommitId(ctx):
...
+    return [
+        {
+            "name": "upload-chromium-cache",
```
Comment (saw-jan) [inline .drone.star:1917]:
```suggestion
            "name": "upload-browsers-cache",
```

File: .drone.star [hunk 9]
```diff
@@ -1899,3 +1910,51 @@ def getOcislatestCommitId(ctx):
...
+            "commands": [
+                "playwright_version=$(bash tests/drone/script.sh get_playwright_version)",
+                "mc alias set s3 $MC_HOST $AWS_ACCESS_KEY_ID $AWS_SECRET_ACCESS_KEY",
+                "mc cp -r -a %s s3/$CACHE_BUCKET/web/$playwright_version/" % dir["chromiumZip"],
+                "mc ls --recursive s3/$CACHE_BUCKET/web",
+            ],
+        },
+    ]
+
+def checkChromiumCache():
```
Comment (saw-jan) [inline .drone.star:1929]:
```suggestion
def checkBrowsersCache():
```

File: .drone.star [hunk 10]
```diff
@@ -1899,3 +1910,51 @@ def getOcislatestCommitId(ctx):
...
+    return [{
+        "name": "check-chromium-cache",
```
Comment (saw-jan) [inline .drone.star:1931]:
```suggestion
        "name": "check-browsers-cache",
```

File: .drone.star [hunk 11]
```diff
@@ -1899,3 +1910,51 @@ def getOcislatestCommitId(ctx):
...
+        "image": MINIO_MC,
+        "environment": minio_mc_environment,
+        "commands": [
+            "mc alias set s3 $MC_HOST $AWS_ACCESS_KEY_ID $AWS_SECRET_ACCESS_KEY",
+            "mc ls --recursive s3/$CACHE_BUCKET/web",
+            "bash tests/drone/script.sh check_chromium_cache",
+        ],
+    }]
+
+def restoreChromiumCache():
```
Comment (saw-jan) [inline .drone.star:1941]:
```suggestion
def restoreBrowsersCache():
```

File: .drone.star [hunk 12]
```diff
@@ -1899,3 +1910,51 @@ def getOcislatestCommitId(ctx):
...
+    return [
+        {
+            "name": "restore-chromium-cache",
```
Comment (saw-jan) [inline .drone.star:1944]:
```suggestion
            "name": "restore-browsers-cache",
```

File: .drone.star [hunk 13]
```diff
@@ -1899,3 +1910,51 @@ def getOcislatestCommitId(ctx):
...
+            "image": MINIO_MC,
+            "environment": minio_mc_environment,
+            "commands": [
+                "playwright_version=$(bash tests/drone/script.sh get_playwright_version)",
+                "mc alias set s3 $MC_HOST $AWS_ACCESS_KEY_ID $AWS_SECRET_ACCESS_KEY",
+                "mc cp -r -a s3/$CACHE_BUCKET/web/$playwright_version/playwright-chromium.tar.gz %s" % dir["web"],
+            ],
+        },
+        {
+            "name": "unzip-chromium-cache",
```
Comment (saw-jan) [inline .drone.star:1954]:
```suggestion
            "name": "unzip-browsers-cache",
```

File: tests/drone/script.sh [hunk 14]
```diff
@@ -29,11 +29,43 @@ check_ocis_cache() {
...
+    if [[ -z "$playwright_version" ]]; then
+        echo "Error: Playwright package not found in package.json." >&2
+        exit 78
+    fi
+
+    echo "$playwright_version"
+}
+
+# Function to check if the cache exists for the given commit ID
+check_chromium_cache() {
```
Comment (saw-jan) [inline tests/drone/script.sh:48]:
```suggestion
check_browsers_cache() {
```

**Review by saw-jan — COMMENTED** [review]
Consider using `browsers` instead of `chromium`

File: .drone.star (hunk 3 above)
Comment (amrita-shrestha) [reply .drone.star:42]: whats the problem with chromium?

Comment (saw-jan) [reply .drone.star:42]: the archive is of folder where browsers are installed by playwright not only chromium
Should we create another path entry for other browsers? like `playwright-firefox.tar.gz`

**Comment by ScharfViktor** [comment]
thank you Amrita, that is great job 👍 
I'm fine with chromium naming if we use only chromium. when we use multi-browser testing in the CI -> we can use abstract name IMHO
I have questions: 
- do we know how long we keep data in the MINIO_MC? what happens if we don't update playwright in for example 30 days and cache will be deleted? should we check that chromium cache is exist?
- in your PR we install chromium https://drone.owncloud.com/owncloud/web/49324/5/5 but we don't update playwright. it happened since we didn't have the cache, correct? in normal case we don't need install chromium when no playwright update

**Comment by amrita-shrestha** [comment]
> thank you Amrita, that is great job 👍 I'm fine with chromium naming if we use only chromium. when we use multi-browser testing in the CI -> we can use abstract name IMHO I have questions:
> 
> * do we know how long we keep data in the MINIO_MC? what happens if we don't update playwright in for example 30 days and cache will be deleted? should we check that chromium cache is exist?
> * in your PR we install chromium https://drone.owncloud.com/owncloud/web/49324/5/5 but we don't update playwright. it happened since we didn't have the cache, correct? in normal case we don't need install chromium when no playwright update

@ScharfViktor, I believe the cache lifespan is 30 days.

 **check_chromium_cache** method verifies the existence of the Chromium cache during each build, based on the Playwright version specified in the **package.json** file. If the cache is missing, the **installChromium** and **cacheChromium** steps are triggered, aligned with the specified Playwright version.


If the bot updates the Playwright version, a new Chromium version will be installed and cached, while the old Chromium cache will remain until its lifespan expires.

I can add a script to remove the old cache when a new one is uploaded, ensuring that multiple cache versions don’t accumulate unnecessarily.

**Review by saw-jan — APPROVED** [review]
:+1:

**Pull request merged**